v2/contribute/decisions/oauth/0002-supported-flows.mdx

```diff
@@ -0,0 +1,58 @@
+---
+id: "0002"
+title: Support non-deprecated flows in rfc6749 + PKCE
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Support non-deprecated flows in rfc6749 + PKCE
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+The OAuth2 docs/rfcs contain a lot of different grant types/flows, we need to decide to which to implement. There are a few groups I'd like to consider:
+- OAuth2.0 (the ones defined in rfc6749) grant types: Authorization Code, Implicit, Resource Owner Password Credentials, Client Credentials
+- Refresh token (defined in rfc6749, but not as a grant type)
+- Authorization Code w/ PKCE (not defined in rfc6749)
+- Other extension grant type (e.g.: Device code)
```

Comment:
Nit: AFAIK, `device code` grant type is not an extension grant type, rather its one among the grant types.
Also , `refresh token` is not a standalone flow/grant type but rather an extension to other grant types.

Reply:
- refresh token is not a grant type (as it's noted above) and technically not a separate flow either. I can note that as well, the main point here was that we need to decide to support refresh tokens or not.
- I meant extension grant type as described here: https://www.rfc-editor.org/rfc/rfc6749#section-4.5